### DISCUSSION TITLE
How to calculate the distance estimation for amino acid?

### DISCUSSION THREAD

**Discussion by lqhungpt** [iqtree/iqtree2#185] [Q&A] [answered]
Hi Admin,
Can you give me the command line or how to estimate the distance matrix for amino acid with JTT model? I want to print out this matrix in output file.
Hung

**Comment by bqminh** [comment, marked as answer]
just run

```
iqtree2 -s ALN_FILE -m JTT
```

IQ-TREE will print a file with suffix ".mldist". This is the distance matrix under JTT model.
Note that this run will also infer a tree. If you only want to get the distance, you can speed up by adding `-te BIONJ` option:

```
iqtree2 -s ALN_FILE -m JTT -te BIONJ
```

It will only compute .mldist and the distanced based BIONJ tree, which is much faster for large alignment.

[!["Buy Me A Coffee"](https://www.buymeacoffee.com/assets/img/custom_images/orange_img.png)](https://www.buymeacoffee.com/bqminh)

**Reply by lqhungpt** [reply]
Thank you, Prof. Bui. Can you show me how to estimate the F84 distance model with aligned nucleotide?
Hung

**Reply by roblanf** [reply]
Hi @lqhungpt

F84 is equivalent to HKY in terms of the parameterisation, i.e. a transition/transversion rate and unequal base frequencies.

So, you can do:

```
iqtree2 -s ALN_FILE -m HKY
```

Another option is to use model selection to find a model which best fits the data (including not just the structure of the Q matrix like GTR or HKY, but also the distribution of rates across sites, like +G, +I+G, +R etc).

To do that, followed by tree estimation:

```
iqtree2 -s ALN_FILE -m MFP
```

Hope that helps,

Rob
